File: benchmark.py
SIZE = 50000  # Define the size of the arrays

# Initialize arrays A, B, and C with zeros
A = [0] * SIZE
B = [0] * SIZE 
C = [0] * SIZE

# Populate arrays A and B
for i in range(SIZE):
    A[i] = i                  # A contains increasing integers from 0 to SIZE-1
    B[i] = 50000 - i          # B contains decreasing integers from SIZE to 1
    C[i] = 0                  # C is initialized to all zeros

# Perform computation: 
for i in range(SIZE):
    for j in range(A[i] + 1): # Loop from 0 to A[i] (inclusive)
        C[i] += B[j]          # Add B[j] to C[i]

# Calculate checksum by summing all elements in C
checksum = sum(C)
print("Checksum:", checksum)

expected = 41667916675000     # Expected checksum value

# Verify result
if checksum != expected:
    print("ERROR: expected", expected)
else:
    print("Result is correct")

File: benchmark.c
#include <stdio.h>
#include <stdlib.h>
#define SIZE 50000

int main() {
    //allocate memory for three arrays of size SIZE
    int A[SIZE], B[SIZE], C[SIZE];

    //initialize arrays A, B, and C with some values
    for (int i = 0; i < SIZE; i++) {
        A[i] = i;
        B[i] = 50000 - i;
        C[i] = 0;
    }
    //compute C[i] as the sum of A[0] to A[i]
    for (int i = 0; i < SIZE; i++) {
        for (int j = 0; j <= A[i]; j++) {
            C[i] += B[j];
        }
    }
    //compute a checksum to verify correctness
    long long checksum = 0;
    for (int i = 0; i < SIZE; i++) {
        checksum += C[i];
    }

    //print the first 10 elements of array C to verify correctness
    for (int i = 0; i < 10; i++) {
        printf("C[%d] = %d\n", i, C[i]);
    }

    const long long expected_checksum = 41667916675000LL;
    printf("Computed checksum: %lld\n", checksum);

    if (checksum != expected_checksum) {
        printf("ERROR: Checksum does not match expected value!\n");
        return 1; // Non-zero exit for error
    }

    return 0;
}